invenio_oaiserver/query.py

```diff
@@ -84,7 +58,7 @@ def get_affected_records(spec=None, search_pattern=None):
         queries.append(Q('match', **{'_oai.sets': spec}))
 
     if search_pattern:
-        queries.append(Query(search_pattern).query.accept(ElasticSearchDSL()))
+        Q('query_string', query=search_pattern)
```

Comment:
This statement is currently doing nothing since it's not added to the ``queries`` (i.e. ``queries.append(....)``).

Also, we should have support for pluggable query parsers, meaning this part should likely be a function that by default uses Elasticsearch, but can be change to use e.g Invenio query parser.

Let's talk IRL about this one.